### ISSUE TITLE
Content validation multi-threading issue

### ISSUE THREAD

**Issue by jordanpadams** [NASA-PDS/validate#167]
**Describe the bug**
While testing for #165 , came across new internal_error being thrown by validate. After adding some stacktraces in, came across the error:
```
java.lang.reflect.InvocationTargetException
...
Caused by: java.util.concurrent.RejectedExecutionException: Task gov.nasa.pds.tools.validate.rule.pds4.LabelInFolderRule$1@9771c71 rejected from java.util.concurrent.ThreadPoolExecutor@63c66980[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 0]
...
	at gov.nasa.pds.tools.validate.rule.pds4.LabelInFolderRule.validateLabelsInFolder(LabelInFolderRule.java:63)
```

---
Update: After poking at this a bit, this just seems like the thread pool is terminating too early for whatever reason. The error appears to occur any time you perform anything but pds4.label validation, which I guess makes sense since all the others call that method.
---

**To Reproduce**
1. Download data from my TBD
2. Execute validate as follows:

```
validate-1.18.0-SNAPSHOT/bin/validate -R pds4.bundle -t /Users/jpadams/test/insight_issue/ --spot-check-data 500
```

The error occurs with or without spot check enabled.

**Full stack trace**
```
java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at gov.nasa.pds.tools.validate.rule.AbstractValidationRule.execute(AbstractValidationRule.java:63)
	at org.apache.commons.chain.impl.ChainBase.execute(ChainBase.java:191)
	at gov.nasa.pds.tools.validate.rule.pds4.SubDirectoryRule.testCollectionDirectories(SubDirectoryRule.java:61)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at gov.nasa.pds.tools.validate.rule.AbstractValidationRule.execute(AbstractValidationRule.java:63)
	at org.apache.commons.chain.impl.ChainBase.execute(ChainBase.java:191)
	at gov.nasa.pds.tools.validate.rule.pds4.SubDirectoryRule.testCollectionDirectories(SubDirectoryRule.java:61)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at gov.nasa.pds.tools.validate.rule.AbstractValidationRule.execute(AbstractValidationRule.java:63)
	at org.apache.commons.chain.impl.ChainBase.execute(ChainBase.java:191)
	at gov.nasa.pds.tools.validate.rule.pds4.SubDirectoryRule.testCollectionDirectories(SubDirectoryRule.java:61)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at gov.nasa.pds.tools.validate.rule.AbstractValidationRule.execute(AbstractValidationRule.java:63)
	at org.apache.commons.chain.impl.ChainBase.execute(ChainBase.java:191)
	at gov.nasa.pds.tools.validate.rule.pds4.SubDirectoryRule.testCollectionDirectories(SubDirectoryRule.java:61)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at gov.nasa.pds.tools.validate.rule.AbstractValidationRule.execute(AbstractValidationRule.java:63)
	at org.apache.commons.chain.impl.ChainBase.execute(ChainBase.java:191)
	at gov.nasa.pds.tools.validate.rule.pds4.SubDirectoryRule.testCollectionDirectories(SubDirectoryRule.java:61)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at gov.nasa.pds.tools.validate.rule.AbstractValidationRule.execute(AbstractValidationRule.java:63)
	at org.apache.commons.chain.impl.ChainBase.execute(ChainBase.java:191)
	at gov.nasa.pds.tools.validate.rule.pds4.CollectionInBundleRule.testCollectionDirectories(CollectionInBundleRule.java:54)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at gov.nasa.pds.tools.validate.rule.AbstractValidationRule.execute(AbstractValidationRule.java:63)
	at org.apache.commons.chain.impl.ChainBase.execute(ChainBase.java:191)
	at gov.nasa.pds.tools.validate.task.ValidationTask.execute(ValidationTask.java:134)
	at gov.nasa.pds.tools.validate.task.BlockingTaskManager.submit(BlockingTaskManager.java:27)
	at gov.nasa.pds.tools.label.LocationValidator.validate(LocationValidator.java:163)
	at gov.nasa.pds.validate.ValidateLauncher.doValidation(ValidateLauncher.java:1226)
	at gov.nasa.pds.validate.ValidateLauncher.processMain(ValidateLauncher.java:1423)
	at gov.nasa.pds.validate.ValidateLauncher.main(ValidateLauncher.java:1466)
Caused by: java.util.concurrent.RejectedExecutionException: Task gov.nasa.pds.tools.validate.rule.pds4.LabelInFolderRule$1@9771c71 rejected from java.util.concurrent.ThreadPoolExecutor@63c66980[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 0]
	at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2063)
	at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:830)
	at java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1379)
	at gov.nasa.pds.tools.validate.rule.pds4.LabelInFolderRule.validateLabelsInFolder(LabelInFolderRule.java:63)
```

Test Data: On pds-dev-el7 alongside the previous test data with the folder name `insight_data_i167`

**Comment by jordanpadams** [comment]
@galenatjpl is there a chance some changes to the `LabelInFolderRule` class snuck into your changes that were maybe not finished? doesn't seem like it is working.

**Comment by jordanpadams** [comment]
you commit was out at 66b549e4501711d1 .

**Comment by jordanpadams** [comment]
Resolved per a9577d84f11eed3bdc1e3b2ed431827a743f231e
